### PR TITLE
Exploratory walkthrough of data ingestion / viz from cancerprof and a basic Shiny app

### PR DESCRIPTION
**Pull request by vsriram24**
Includes R code in QMD files to explore data ingestion, munging, and visualization (using {leaflet} and {tmap}) for cancer incidence data. Also includes directories of publicly-sourced shape files for geospatial data visualization.

Next steps are to comment / clean code as well as remove any hard-coded elements, and then port this over into a Shiny app so that data visualization can be explored further.

### CODE REVIEW

**Comment by vsriram24** [comment]
This branch also includes the creation of a basic Shiny app.

File: R/app.R [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      race = input$race
+      sex = input$sex
+      age = input$age
+      stage = input$stage
+      year = input$year
+
+      # Create a new DuckDB corresponding to incidence data from US states
+      con <- dbConnect(duckdb::duckdb(), "cancer-incidence-usa-state.duckdb")
+
+      ## 1. Ingest data from State Cancer Profiles if needed
```
Comment (seankross) [inline R/app.R:166]: Data ingestion and cleaning should take place outside of the shiny app itself, the app logic should assume that the data is completely prepared for the app. This make it easier to iterate on data cleaning without necessarily touching the app.

File: R/helper_functions.R [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+#'    "all cancer sites",
+#'    "All Races (includes Hispanic)",
+#'    "both sexes",
+#'    "all ages",
+#'    "all stages",
+#'    "latest 5 year average"
+#' )
+#'
+
+get_incidence_db_name <- function(cancer, race, sex, age, stage, year){
```
Comment (seankross) [inline R/helper_functions.R:29]: We should talk about this database schema, I'm not convinced this is the best way forward.

**Comment by seankross** [comment]
Generally speaking this is a great start. Both of my main comments are pretty high level, let's chat about it when you get the chance.

**Comment by vsriram24** [comment]
@seankross data ingestion now exists separately from the Shiny app. The latest updates on this branch include a separate pipeline that ingests SCP Washington state county-level data and puts it into the Fred Hutch postgreSQL database. Let me know your feedback on this PR when you have the chance.

**Comment by seankross** [comment]
Hey @vsriram24, unfortunately I am just now getting to this. Which files do I need to run in what order to get the Shiny app to work?

**Comment by vsriram24** [comment]
Hi @seankross, for this branch, there's no need to worry about the Shiny app. You can just run `R/ingest-scp.R` to evaluate the data ingestion pipeline. Note that you will have to create a `.Renviron` file within the `oncarto/` directory that contains the postgreSQL database information for data saving the work. I'll include a comment in the `baseline-app` PR to explain how to run the Shiny app.

**Comment by seankross** [comment]
@vsriram24 in that case I am only going to give feedback on `R/ingest-scp.R` in this PR unless you want feedback on other changes here.

**Review by seankross — APPROVED** [review]

**Comment by seankross** [comment]
@vsriram24 merge this whenever you want

**Pull request merged**